src/Components/Buttons/CatalogButtons.tsx

```diff
@@ -75,35 +73,42 @@ class CatalogButtons extends Component<Props,ICatalogButtons>{
      * @param isCalendarType
      * @param date
      */
+        // TODO: If choosing different version from different button, should old button version jump back to default or stay where it was?
+        //  F.e. if clicking from ICD10-GM-2014 to SwissDRG V9.0, should version button of icd stay at GM-2014 or jump back to GM-2022?
     updateButton = (version, btn, isCalendarType, date) => {
-        if (version === '') {
-            version = this.getVersion(btn);
-        }
-        switch (btn) {
-            case 'ICD':
-                this.setState({lastICD: version});
-                break;
-            case 'SwissDRG':
-                this.setState({lastDRG: version});
-                break;
-            case 'CHOP':
-                this.setState({lastCHOP: version});
-                break;
-            case 'TARMED':
-                this.setState({lastTARMED: version});
-                break;
-            default:
-        }
+        // If empty version, get it using btn.
+        // Set
+        let selectedVersion;
         if (isCalendarType) {
-            version = ''
+            selectedVersion = ''
+        } else {
+            selectedVersion = version === '' ? this.getVersionFromButton(btn) : version;
+            // Update currentVersion.
+            // TODO: are the 'break' statements needed?
+            switch (btn) {
+                case 'ICD':
+                    this.setState({currentICD: version});
```

Comment:
You could do this dynamic:
```
h = {}
h['current' + btn] = version
this.setState(h)
```
Of course you'd have to rename currentDRG to currentSwissDRG.